### ISSUE TITLE
Emacs bindings for Irrlicht/formspecs Textareas

### ISSUE THREAD

**Issue by swagtoy** [minetest/minetest#15154]
### Problem

I have a nasty habit with Emacs bindings. They are primed onto my brain and I can't get away from them.

Lots of software, even GTK and such, offer some sort of basic Emacs bindings. I'm sure even some of US use Emacs bindings since we... well, might write Lua mods in Emacs (well, I use QT Creator these days.. with emacs bindings)

There are lots of mods that offer full textboxes (usually to write ACTUAL Lua code, oddly enough) and even simple ones. I think this would just be a cool "quality of life" thing, and consistancy with stuff like Gettext and GTK and GNU Emacs which already supports Emacs bindings.

### Solutions

Add basic bindings like Ctrl-{A,E}, Ctrl-{<,>}, Ctrl-W, {Alt,Ctrl}-V, {Alt,Ctrl}-{F,B,P,N}, etc etc etc; as an option you can set in the engine.

### Alternatives

Don't add Emacs bindings

### Additional context

M-x Butterfly

**Comment by swagtoy** [comment]
We can probably look for GTK's config file, then look for `gtk-key-theme-name = Emacs` within `~/.config/gtk-3.0/settings.ini`. It looks like gtk4 uses Gconf though, which is a no-go...

**Comment by v-rob** [comment]
I do not think we should support either Emacs or Vi(m) bindings, even as a setting. It's simply too esoteric to have to maintain that. However, we definitely should support more keyboard shortcuts that standard GUIs like Windows, GTK, Qt, etc. all support. For instance, Ctrl+Right is standard in most GUIs to move forwards one word, but we don't support that.

(Also, Vim's modal editing is clearly superior to any other editor. Not a good reason to include it in Minetest, but just sayin'.)

**Comment by swagtoy** [comment]
It's only slightly esoteric, but I agree, we should REALLY do the more traditional keybinds first.

But still, I feel like some basic Emacs bindings couldn't hurt on the side, like C-{f,b,p,n}. It's... not that esoteric, as every GNU Gettext program supports these by default and some software does as well, Even macOS supports Emacs bindings in software!! Clear sign it's not too esoteric :^)

I don't like custom keybindings for that though, I don't think someone will actually want to customize, say, Ctrl+right functionality or add their own bindings for formspecs.

**Comment by swagtoy** [comment]
Yeah, i guess the target audience wouldn't think of it much. Again though, it's one of those "harmless" things that someone could just slip in there.

Control + left/right support would be a better priority. Maybe I should rename this issue and edit it?

**Comment by hlqkj** [comment]
> Add (inferior) Vi-style bindings.. but seriously, we don't need those, unless someone here can pitch a valid reason why they'd ever want that (If you're insane).

Hey, I'm that insane you're talking about 🤣
